ModuleNotFoundError when trying to run pytest

* jointly version: master
* Python version: 3.9.7
* Operating System: Ubuntu (inside Windows Subsystem for Linux)

### Description

Trying to test the package while doing the review for PyOpenSci (https://github.com/pyOpenSci/software-review/issues/45) results in different errors

### What I Did

- Create a conda environment: `conda create -n jointly python poetry`
- Clone the project from github.
- `poetry install; pre-commit install`
- Tried to run all tests with `CONTRIBUTING.md` instructions `py.test tests.test_jointly` failed: 

```
(jointly) asaez@ap-5cg1140pq3:~/pyopensci/jointly$  py.test tests.test_jointly
================================================= test session starts ==================================================
platform linux -- Python 3.9.7, pytest-6.2.4, py-1.10.0, pluggy-0.13.1
rootdir: /home/asaez/pyopensci/jointly
collected 0 items

================================================ no tests ran in 0.00s =================================================
ERROR: file or directory not found: tests.test_jointly
```

- Tried to run all tests from `tests` folder and got this error

```
(jointly) asaez@ap-5cg1140pq3:~/pyopensci/jointly/tests$ py.test
================================================= test session starts ==================================================
platform linux -- Python 3.9.7, pytest-6.2.4, py-1.10.0, pluggy-0.13.1
rootdir: /home/asaez/pyopensci/jointly/tests
collected 5 items / 4 errors / 1 selected

======================================================== ERRORS ========================================================
___________________________________________ ERROR collecting test_helpers.py ___________________________________________
ImportError while importing test module '/home/asaez/pyopensci/jointly/tests/test_helpers.py'.
Hint: make sure your test modules/packages have valid Python names.
Traceback:
../../../anaconda3/envs/jointly/lib/python3.9/importlib/__init__.py:127: in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
test_helpers.py:20: in <module>
    from tests.parquet_reader import get_parquet_test_data
E   ModuleNotFoundError: No module named 'tests.parquet_reader'
______________________________________ ERROR collecting test_helpers_plotting.py _______________________________________
ImportError while importing test module '/home/asaez/pyopensci/jointly/tests/test_helpers_plotting.py'.
Hint: make sure your test modules/packages have valid Python names.
Traceback:
../../../anaconda3/envs/jointly/lib/python3.9/importlib/__init__.py:127: in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
test_helpers_plotting.py:9: in <module>
    from tests.parquet_reader import get_parquet_test_data
E   ModuleNotFoundError: No module named 'tests.parquet_reader'
_______________________________________ ERROR collecting test_shake_extractor.py _______________________________________
ImportError while importing test module '/home/asaez/pyopensci/jointly/tests/test_shake_extractor.py'.
Hint: make sure your test modules/packages have valid Python names.
Traceback:
../../../anaconda3/envs/jointly/lib/python3.9/importlib/__init__.py:127: in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
test_shake_extractor.py:5: in <module>
    from tests.parquet_reader import get_parquet_test_data
E   ModuleNotFoundError: No module named 'tests.parquet_reader'
________________________________________ ERROR collecting test_synchronizer.py _________________________________________
ImportError while importing test module '/home/asaez/pyopensci/jointly/tests/test_synchronizer.py'.
Hint: make sure your test modules/packages have valid Python names.
Traceback:
../../../anaconda3/envs/jointly/lib/python3.9/importlib/__init__.py:127: in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
test_synchronizer.py:11: in <module>
    from tests.parquet_reader import get_parquet_test_data
E   ModuleNotFoundError: No module named 'tests.parquet_reader'
=============================================== short test summary info ================================================
ERROR test_helpers.py
ERROR test_helpers_plotting.py
ERROR test_shake_extractor.py
ERROR test_synchronizer.py
!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!! Interrupted: 4 errors during collection !!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!
```

- Tried to run the tests from the root folder and got the same error:
```
(jointly) asaez@ap-5cg1140pq3:~/pyopensci/jointly$ pytest
================================================= test session starts ==================================================
platform linux -- Python 3.9.7, pytest-6.2.4, py-1.10.0, pluggy-0.13.1
rootdir: /home/asaez/pyopensci/jointly
collected 5 items / 4 errors / 1 selected

======================================================== ERRORS ========================================================
________________________________________ ERROR collecting tests/test_helpers.py ________________________________________
ImportError while importing test module '/home/asaez/pyopensci/jointly/tests/test_helpers.py'.
Hint: make sure your test modules/packages have valid Python names.
Traceback:
../../anaconda3/envs/jointly/lib/python3.9/importlib/__init__.py:127: in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
tests/test_helpers.py:20: in <module>
    from tests.parquet_reader import get_parquet_test_data
E   ModuleNotFoundError: No module named 'tests.parquet_reader'
___________________________________ ERROR collecting tests/test_helpers_plotting.py ____________________________________
ImportError while importing test module '/home/asaez/pyopensci/jointly/tests/test_helpers_plotting.py'.
Hint: make sure your test modules/packages have valid Python names.
Traceback:
../../anaconda3/envs/jointly/lib/python3.9/importlib/__init__.py:127: in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
tests/test_helpers_plotting.py:9: in <module>
    from tests.parquet_reader import get_parquet_test_data
E   ModuleNotFoundError: No module named 'tests.parquet_reader'
____________________________________ ERROR collecting tests/test_shake_extractor.py ____________________________________
ImportError while importing test module '/home/asaez/pyopensci/jointly/tests/test_shake_extractor.py'.
Hint: make sure your test modules/packages have valid Python names.
Traceback:
../../anaconda3/envs/jointly/lib/python3.9/importlib/__init__.py:127: in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
tests/test_shake_extractor.py:5: in <module>
    from tests.parquet_reader import get_parquet_test_data
E   ModuleNotFoundError: No module named 'tests.parquet_reader'
_____________________________________ ERROR collecting tests/test_synchronizer.py ______________________________________
ImportError while importing test module '/home/asaez/pyopensci/jointly/tests/test_synchronizer.py'.
Hint: make sure your test modules/packages have valid Python names.
Traceback:
../../anaconda3/envs/jointly/lib/python3.9/importlib/__init__.py:127: in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
tests/test_synchronizer.py:11: in <module>
    from tests.parquet_reader import get_parquet_test_data
E   ModuleNotFoundError: No module named 'tests.parquet_reader'
=============================================== short test summary info ================================================
ERROR tests/test_helpers.py
ERROR tests/test_helpers_plotting.py
ERROR tests/test_shake_extractor.py
ERROR tests/test_synchronizer.py
!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!! Interrupted: 4 errors during collection !!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!
============================================ 4 errors in 135.08s (0:02:15) =============================================
```

--- a/CONTRIBUTING.rst
+++ b/CONTRIBUTING.rst
@@ -1,4 +1,5 @@
 .. highlight:: shell
+.. _contributing_label:
 
 ============
 Contributing
@@ -6,6 +7,9 @@ Contributing
 
 Contributions are welcome, and they are greatly appreciated! Every little bit
 helps, and credit will always be given.
+
+We would appreciate a feature suggestion issue before you create a PR so we can
+discuss the feature, its use, and its implementation.
 
 You can contribute in many ways:
 
@@ -107,12 +111,12 @@ Tips
 
 To run a subset of tests::
 
-    $ py.test tests.test_jointly
+    $ py.test tests/test_examples.py
 
 
 To run all tests::
 
-    $ py.test tests.test_jointly
+    $ py.test
 
 Deploying
 ---------
@@ -120,5 +124,6 @@ Deploying
 A reminder for the maintainers on how to deploy.
 Make sure all your changes are committed, including an entry in HISTORY.rst and an update of the old version code
 in ``docs/conf.py`` and ``pyproject.toml``.
+Please also link the PR in your history entry.
 
 GitHub will then deploy to PyPI if tests pass.

--- a/HISTORY.rst
+++ b/HISTORY.rst
@@ -2,6 +2,13 @@
 History
 =======
 
+1.0.4 (2021-12-27)
+------------------
+
+* fix contributing.rst guide to testing
+* improve documentation problems
+* c.f. https://github.com/hpi-dhc/jointly/pull/12
+
 1.0.3 (2021-12-09)
 ------------------
 

--- a/README.rst
+++ b/README.rst
@@ -54,6 +54,8 @@ Install the package from pypi:
 
     pip install jointly
 
+You might want to check out our :ref:`contributing_label` guide in case you want to edit the package.
+
 Usage
 -----
 

--- a/docs/reference.rst
+++ b/docs/reference.rst
@@ -43,6 +43,14 @@ jointly.helpers
    :undoc-members:
    :show-inheritance:
 
+jointly.helpers_plotting
+---------------
+
+.. automodule:: jointly.helpers_plotting
+   :members:
+   :undoc-members:
+   :show-inheritance:
+
 jointly.abstract\_extractor
 ---------------------------
 
@@ -56,15 +64,6 @@ jointly.log
 -----------
 
 .. automodule:: jointly.log
-   :members:
-   :undoc-members:
-   :show-inheritance:
-
-
-jointly.segment\_selector
--------------------------
-
-.. automodule:: jointly.segment_selector
    :members:
    :undoc-members:
    :show-inheritance:

--- a/jointly/helpers.py
+++ b/jointly/helpers.py
@@ -1,4 +1,4 @@
-"""Contains plotting helpers"""
+"""Contains various helper functions useful in conjunction with or internally to jointly."""
 import logging
 from pprint import pprint
 from typing import List, Tuple, Iterable
@@ -15,7 +15,12 @@ logger = logging.getLogger("jointly.helpers")
 def calculate_magnitude(
     df: pd.DataFrame, of_cols: List[str], title: str = "Magnitude"
 ) -> pd.DataFrame:
-    """Calculate the magnitude of a subset of columns from a DataFrame"""
+    """
+    Calculate the magnitude of a subset of columns from a DataFrame.
+
+    Will return 0 if the input data is NaN to allow future algorithms
+    to continue working despite the NaN in the input values.
+    """
     data = df[of_cols]
     result = np.sqrt(np.square(data).sum(axis=1))
     result.name = title
@@ -107,6 +112,9 @@ def get_stretch_factor(
     Get the stretch factor required to stretch the duration between segments such that it will fit exactly to the
     signal when shifted by the amount given by timeshifts.
 
+    This is a function that should basically exclusively be used within jointly, as the parameters are produced during
+    the synchronization process.
+
     :param segments: the segment instance containing the segment info to be stretched
     :param timeshifts: the timeshifts that should be applied to make the signal align to the reference signal
     :return: a float as described above
@@ -135,6 +143,10 @@ def verify_segments(signals: Iterable[str], segments: SyncPairs):
 def get_segment_data(
     dataframe: pd.DataFrame, segments: SyncPairs, col: str, segment: str
 ) -> Tuple[pd.Timestamp, pd.Timestamp, pd.DataFrame]:
+    """
+    Return a 3-tuple of start and end indices plus data
+    within that timeframe of the given column in the given dataframe.
+    """
     start = segments[col][segment]["start"]
     end = segments[col][segment]["end"]
     return start, end, dataframe[col][start:end]

--- a/pyproject.toml
+++ b/pyproject.toml
@@ -1,6 +1,6 @@
 [tool.poetry]
 name = "jointly"
-version = "1.0.3"
+version = "1.0.4"
 description = "Synchronize sensor data from accelerometer shakes"
 authors = [
     "Ariane Morassi Sasso <ariane.morassi-sasso@hpi.de>",

